### PR TITLE
fix: security issues sonar exception

### PR DESCRIPTION
**Pull request by nirinaHelena**

### CODE REVIEW

File: src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.zip.ZipOutputStream;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class FileUnzipperTest extends FacadeIT {
+  @Autowired private FileUnzipper fileUnzipper;
+
+  private static final String FILE_NAME =
+      "src"
+          + File.separator
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java]: well, no. just "/" is enough

File: src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  ZipFile zipFile() throws IOException {
+    File zipFile = File.createTempFile("zip-file-" + now(), ZIP_FILE_SUFFIX);
+    try (FileOutputStream fos = new FileOutputStream(zipFile);
+        ZipOutputStream zos = new ZipOutputStream(fos)) {
+      zos.putNextEntry(new ZipEntry(".."));
+      zos.finish();
+      zos.flush();
+      return new ZipFile(zipFile);
+    } catch (IOException e) {
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java]: useless. you already throws IOException in your signature

File: src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          + File.separator
+          + "test"
+          + File.separator
+          + "resources"
+          + File.separator
+          + "mockData"
+          + File.separator
+          + "zipFile.zip";
+
+  ZipFile zipFile() throws IOException {
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileUnzipperTest.java]: bad name. You are not constructing a standard zipFile. You are constructing a MALICIOUS zipFile (which contains path traversal attack with ".."). Make it explicit in the name

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  @BeforeEach
+  void setUp() {
+    extensionGuesser = mock(ExtensionGuesser.class);
+    fileWriter = new FileWriter(extensionGuesser);
+  }
+
+  @Test
+  void testApplyThrowsApiExceptionOnIOException() {
+    byte[] bytes = new byte[] {1, 2, 3};
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: var

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    File directory = null;
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: bad. use null directly where needed

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        .thenThrow(new IOException());
+
+    assertThrows(
+        app.bpartners.geojobs.model.exception.ApiException.class,
+        () -> fileWriter.apply(bytes, directory));
+  }
+
+  @Test
+  void testWriteThrowsIllegalArgumentExceptionForInvalidDirectory() {
+    byte[] bytes = new byte[] {1, 2, 3};
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: var

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    File invalidDirectory = new File("../");
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: var

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    String filename = "testfile";
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: var

File: src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+    assertThrows(
+        IllegalArgumentException.class, () -> fileWriter.write(bytes, invalidDirectory, filename));
+  }
+
+  @Test
+  void testWriteThrowsDirectoryNull() {
+    byte[] bytes = new byte[] {1, 2, 3};
```
Comment (hei-teacher) [inline src/test/java/app/bpartners/geojobs/unit/FileWriterTest.java]: var everywhere possible

**Review by hei-teacher — CHANGES_REQUESTED** [review]

File: src/main/java/app/bpartners/geojobs/file/FileWriter.java [hunk 10]
```diff
@@ -41,6 +41,9 @@ public File apply(byte[] bytes, @Nullable File directory) {
   }
 
   public File write(byte[] bytes, @Nullable File directory, String filename) {
+    if (directory != null && directory.getName().contains("..")) {
+      throw new IllegalArgumentException("name must not contain .. but receceived: pathValue");
```
Comment (nvagno) [inline src/main/java/app/bpartners/geojobs/file/FileWriter.java:45]: what is `pathValue`?

**Review by nvagno — CHANGES_REQUESTED** [review]

**Review by nvagno — APPROVED** [review]

**Pull request merged**